lib/ansible/modules/cloud/vmware/vsphere_copy.py

```diff
@@ -153,13 +173,22 @@ def main():
     validate_certs = module.params.get('validate_certs')
     timeout = module.params.get('timeout')
 
-    fd = open(src, "rb")
-    atexit.register(fd.close)
+    try:
+        fd = open(src, "rb")
+        atexit.register(fd.close)
+    except Exception as e:
+        module.fail_json(msg="Failed to open src file %s" % to_native(e))
 
-    data = mmap.mmap(fd.fileno(), 0, access=mmap.ACCESS_READ)
-    atexit.register(data.close)
+    if os.stat(src).st_size == 0:
+        data = ''
+    else:
+        data = mmap.mmap(fd.fileno(), 0, access=mmap.ACCESS_READ)
```

Comment:
Just a thought: I'm pretty sure we can avoid the `mmap` and just pass the fd through `open_url()`'s data parameter.